File: src/com/eagle/programmar/CSS/Values/CSS_NumericValue.java
// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Aug 12, 2022

package com.eagle.programmar.CSS.Values;

import com.eagle.programmar.CSS.Terminals.CSS_KeywordChoice;
import com.eagle.programmar.CSS.Terminals.CSS_Number;
import com.eagle.programmar.CSS.Terminals.CSS_Punctuation;
import com.eagle.tokens.TokenChooser;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.punctuation.PunctuationSlash;

public class CSS_NumericValue extends TokenSequence
{
	public @S(10) CSS_Number number;
	public @S(20) @OPT CSS_NumericSuffix suffix;
	public @S(30) @OPT CSS_SlashNumber slash;
	
	public static class CSS_NumericSuffix extends TokenChooser
	{
		public @CHOICE CSS_KeywordChoice units = new CSS_KeywordChoice(
				"deg",
				"em",
				"in",
				"px",
				"s",
				"vw"
		);
		public @CHOICE CSS_Punctuation percent = new CSS_Punctuation('%');
	}
	
	public static class CSS_SlashNumber extends TokenSequence
	{
		public @S(10) PunctuationSlash slash;
		public @S(20) CSS_Number number;
		public @S(30) @OPT CSS_NumericSuffix suffix;
	}
}

File: csrc/com/eagle/programmar/CSS/Values/CSS_NumericValue.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Aug 12, 2022

namespace com.eagle.programmar.CSS.Values
{
	using CSS_KeywordChoice = com.eagle.programmar.CSS.Terminals.CSS_KeywordChoice;
	using CSS_Number = com.eagle.programmar.CSS.Terminals.CSS_Number;
	using CSS_Punctuation = com.eagle.programmar.CSS.Terminals.CSS_Punctuation;
	using TokenChooser = com.eagle.tokens.TokenChooser;
	using TokenSequence = com.eagle.tokens.TokenSequence;
	using PunctuationSlash = com.eagle.tokens.punctuation.PunctuationSlash;

	public class CSS_NumericValue : TokenSequence
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.CSS.Terminals.CSS_Number number;
		public CSS_Number number;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT CSS_NumericSuffix suffix;
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) @OPT CSS_SlashNumber slash;
		public  OPT;

		public class CSS_NumericSuffix : TokenChooser
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE CSS_KeywordChoice XXunits = new com.eagle.programmar.CSS.Terminals.CSS_KeywordChoice("deg", "em", "in", "px", "s", "vw");
			public CSS_KeywordChoice XXunits = new CSS_KeywordChoice("deg", "em", "in", "px", "s", "vw");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE CSS_Punctuation XXpercent = new com.eagle.programmar.CSS.Terminals.CSS_Punctuation('%');
			public CSS_Punctuation XXpercent = new CSS_Punctuation('%');
		}

		public class CSS_SlashNumber : TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.tokens.punctuation.PunctuationSlash slash;
			public PunctuationSlash slash;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.CSS.Terminals.CSS_Number number;
			public CSS_Number number;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) @OPT CSS_NumericSuffix suffix;
			public  OPT;
		}
	}
}
